### PR TITLE
Add documentation on how to be a good buddy

### PR DESCRIPTION
**Pull request by owenatgov**
## What
Adds documentation on how to be a good "buddy" as part of a new joiner's onboarding process.

## Why
The buddy convention is something we follow stringently in the frontend community. Following a presentation I did to said community, it was proposed some documentation to the effect of that presentation could be added to these dev docs, potentially as a companion piece to our docs on the [responsibilities of a tech lead](https://docs.publishing.service.gov.uk/manual/tech-lead-responsibilities.html). This is an attempt to capture the contents of that presentation which went into recent positive examples of being a buddy and the differences between other roles like line manager to a new joiner.

### CODE REVIEW

File: source/manual/how-to-be-a-good-buddy.html.md
Comment (maxgds) [inline source/manual/how-to-be-a-good-buddy.html.md:21]: Interesting, I was under the impression we looked for on-programme but off-team for frontenders. Guess I was wrong! With many teams only having 1 FE it's hard to tell!

Comment (owenatgov) [reply source/manual/how-to-be-a-good-buddy.html.md:21]: That's a good point. I was under the impression that what I wrote was the ideal but that's also my best guess. I'm open to changing this, do we have prior art/thoughts on this?

Comment (maxgds) [reply source/manual/how-to-be-a-good-buddy.html.md:21]: I'm told backend aim for someone in team, so that's probably worth sticking with.

File: source/manual/how-to-be-a-good-buddy.html.md
Comment (ChrisBAshton) [inline source/manual/how-to-be-a-good-buddy.html.md:10]:
```suggestion
When a new joiner starts on govuk, it's good practice to assign them a "buddy" to help them get started on GOV.UK, explain its infrastructure and provide initial support.
```

File: source/manual/how-to-be-a-good-buddy.html.md
Comment (ChrisBAshton) [inline source/manual/how-to-be-a-good-buddy.html.md:14]: This duplicates some of the content above:

```suggestion
A buddy acts as a temporary guide for new joiners. Their responsibilities include:
```

File: source/manual/how-to-be-a-good-buddy.html.md
Comment (ChrisBAshton) [inline source/manual/how-to-be-a-good-buddy.html.md:19]:
```suggestion
- generally making them feel welcome
```

File: source/manual/how-to-be-a-good-buddy.html.md
Comment (ChrisBAshton) [inline source/manual/how-to-be-a-good-buddy.html.md:29]: Whilst I agree with the sentiment, I suggest neutralising to this:

```suggestion
- handle general admin, HR processes and SOP
```

File: source/manual/how-to-be-a-good-buddy.html.md
Comment (ChrisBAshton) [inline source/manual/how-to-be-a-good-buddy.html.md:33]: Link to https://docs.publishing.service.gov.uk/manual/get-started.html?

(I did think perhaps to https://trello.com/b/yMWYSGlz/govuk-developer-onboarding instead, though this may still be awaiting sign-off from Senior Tech. And in any case, that feels more like a line manager's responsibility)

File: source/manual/how-to-be-a-good-buddy.html.md
Comment (ChrisBAshton) [inline source/manual/how-to-be-a-good-buddy.html.md:34]:
```suggestion
- Schedule catchups with them in their first week or two, to talk through things like our stack and our tooling
```

File: source/manual/how-to-be-a-good-buddy.html.md
Comment (ChrisBAshton) [inline source/manual/how-to-be-a-good-buddy.html.md:37]:
```suggestion
- Consider having a handover discussion after the first month to check how the new joiner found onboarding, then feed that feedback into our general onboarding process
```

File: source/manual/how-to-be-a-good-buddy.html.md
Comment (ChrisBAshton) [inline source/manual/how-to-be-a-good-buddy.html.md:38]: This advice feels a bit unofficial to be included in this otherwise official-looking list. I'd consider closing with something like this instead:

```suggestion

There's no official end date for being a buddy. It is up to you and the new joiner to continue checking in for as long as you both feel it's useful.
```

**Review by ChrisBAshton — COMMENTED** [review]
Thanks for writing this up - it fills a bit of a knowledge gap! I've done a quick drive-by review below.

As the nature of buddying is (loosely) formal, it does feel like this should get a ✅  from someone in senior tech before merging, IMO, so that it's not just yours & my opinions. 😀

File: source/manual/how-to-be-a-good-buddy.html.md
Comment (owenatgov) [reply source/manual/how-to-be-a-good-buddy.html.md:29]: Whoops. I put that in my original presentation as a little joke and completely forgot I'd left it in. Not my most professional piece of content 😅

**Comment by owenatgov** [comment]
@ChrisBAshton Cheers for the review and good shout. Have posted in govuk-tech-leads on slack.

File: source/manual/how-to-be-a-good-buddy.html.md
Comment (barrucadu) [inline source/manual/how-to-be-a-good-buddy.html.md:29]: Should there be something after this "and"?

File: source/manual/how-to-be-a-good-buddy.html.md
Comment (kevindew) [inline source/manual/how-to-be-a-good-buddy.html.md:10]:
```suggestion
When a new joiner starts on GOV.UK, it's good practice for the team they are joining to assign them a "buddy" to help them get started on GOV.UK, explain it's infrastructure and provide initial support.
```

Just to resolve an ambiguity about who assigns buddy.

File: source/manual/how-to-be-a-good-buddy.html.md
Comment (kevindew) [inline source/manual/how-to-be-a-good-buddy.html.md:14]:
```suggestion
A buddy acts as a temporary guide for new joiners. It's an informal role and their responsibilities include:
```

I kind of wanted to get the phrase "informal role" in somewhere and thought this was best place.

**Review by kevindew — COMMENTED** [review]
Aside from the lingering "and" that Michael spotted this looks great. Thanks for putting it together 🏅

File: source/manual/how-to-be-a-good-buddy.html.md
Comment (owenatgov) [reply source/manual/how-to-be-a-good-buddy.html.md:29]: That should've been "and SOP" but I'm now wondering if it's useful to include SOP in this. I'll amend.

**Comment by chao-xian** [comment]
This is awesome, thank you @owenatgov! I would add that for some teams, some of this stuff might be covered by a Tech Lead doing some induction too but there's no harm in there being repetition and cross checking that these things did happen. In that spirit I would like it emphasised that it's _everyone's_ responsibility to "be a buddy" and that the assigned Buddy is just a semi-official one.

**Review by kevindew — APPROVED** [review]
Thanks again

**Comment by owenatgov** [comment]
@chao-xian Do we think it's worth adding a line at the bottom to this effect re: everyone being welcoming?

**Pull request merged**
